components/apimgt/org.wso2.carbon.apimgt.eventing.hub/src/main/java/org/wso2/carbon/apimgt/eventing/hub/EventHubEventPublisherFactory.java

```diff
@@ -0,0 +1,77 @@
+import org.wso2.carbon.apimgt.eventing.EventPublisherFactory;
+import org.wso2.carbon.apimgt.eventing.hub.internal.ServiceReferenceHolder;
+import org.wso2.carbon.event.output.adapter.core.OutputEventAdapterConfiguration;
+import org.wso2.carbon.event.output.adapter.core.exception.OutputEventAdapterException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Factory for getting event hub event publisher instances.
+ */
+public class EventHubEventPublisherFactory implements EventPublisherFactory {
+    private static final Log log = LogFactory.getLog(EventHubEventPublisherFactory.class);
+
+    @Override
+    public void configure(Map<String, String> configuration) {
+        if (ServiceReferenceHolder.getInstance().isConfiguredOutputEventAdapterService()) {
+            if (configuration != null) {
+                OutputEventAdapterConfiguration adapterConfiguration = new OutputEventAdapterConfiguration();
+                adapterConfiguration.setName(EventHubPublisherConstants.EVENT_HUB_NOTIFICATION_EVENT_PUBLISHER);
+                adapterConfiguration.setType(EventHubPublisherConstants.BLOCKING_EVENT_TYPE);
+                adapterConfiguration.setMessageFormat(EventHubPublisherConstants.BLOCKING_EVENT_FORMAT);
+                Map<String, String> adapterParameters = new HashMap<>();
+                adapterParameters.put(EventHubPublisherConstants.RECEIVER_URL,
+                        configuration.get(EventHubPublisherConstants.RECEIVER_URL));
+                adapterParameters.put(EventHubPublisherConstants.AUTHENTICATOR_URL,
+                        configuration.get(EventHubPublisherConstants.AUTHENTICATOR_URL));
+                adapterParameters.put(EventHubPublisherConstants.USERNAME,
+                        configuration.get(EventHubPublisherConstants.USERNAME));
+                adapterParameters.put(EventHubPublisherConstants.PASSWORD,
+                        configuration.get(EventHubPublisherConstants.PASSWORD));
+                adapterParameters.put(EventHubPublisherConstants.PROTOCOL,
+                        configuration.get(EventHubPublisherConstants.PROTOCOL));
+                adapterParameters.put(EventHubPublisherConstants.PUBLISHING_MODE,
+                        EventHubPublisherConstants.NON_BLOCKING);
+                adapterParameters.put(EventHubPublisherConstants.PUBLISHING_TIME_OUT, "0");
+                adapterConfiguration.setStaticProperties(adapterParameters);
+                try {
+                    ServiceReferenceHolder.getInstance().getOutputEventAdapterService().create(adapterConfiguration);
+                    ServiceReferenceHolder.getInstance().setConfiguredOutputEventAdapterService(true);
+                } catch (OutputEventAdapterException e) {
+                    log.warn("Exception occurred while creating WSO2 Event Adapter. Event notification may not work "
+                            + "properly", e);
+                }
+            } else {
```

Comment:
I don't think we should do a null check. This cannont be null the one who calls the method needs to ensure that.